frontend/src/RecipeCard/RecipeCard.jsx

```diff
@@ -20,19 +59,14 @@ const handleClick = () => {
                 <h2 className = "recipeName" > {props.title}</h2>
                 <div className='underName'>
                 <h3 className= "cuisine"> Cuisines:</h3>
-
-
                 <div className= "favRecipe">
-                    <a className= "favoriteIcon" onClick={(e) => {
-                            e.stopPropagation();
-                            setIsFav(!isFav);
-                            }}>
-                        {isFav ?
-                            <i className="fa-solid fa-heart"></i> :
-                            <i className="fa-regular fa-heart"></i>
-                    }
-                    </a>
+                <a className="favoriteIcon" onClick={handleAddFavorite}>
 
+                    {isFav ?
```

Comment:
only the className is changing here based on `isFav`: you can do a ternary operator only for the className and not an element itself, smth like: `<i className={isFav? "fa-solid fa-heart" : "other-class-name"}></I>`
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator